src/connector/src/sink/iceberg/mod.rs

```diff
@@ -311,9 +318,64 @@ impl IcebergSink {
                 }
             };
 
+            let partition_fields = match &self.config.partition_by {
+                Some(partition_field) => {
+                    let mut partition_fields = Vec::<UnboundPartitionField>::new();
+                    // captures column, transform(column), transform(n,column), transform(n, column)
+                    let re = Regex::new(
+                        r"(?<transform>\w+)(\(((?<n>\d+)?(?:,|(,\s)))?(?<field>\w+)\))?",
+                    )
+                    .unwrap();
+                    if !re.is_match(partition_field) {
+                        bail!(format!("Invalid partition fields: {}", partition_field))
+                    }
+                    let caps = re.captures_iter(partition_field);
+                    for mat in caps {
+                        let (column, transform) = if mat["n"].is_empty() && mat["field"].is_empty()
+                        {
+                            (&mat["func"], Transform::Identity)
+                        } else {
+                            let mut func = mat["func"].to_owned();
+                            let n = &mat["n"];
+                            if func == "bucket" || func == "truncate" {
+                                func = format!("{func}({n})");
+                            }
+                            (&mat["field"], Transform::from_str(&func).unwrap())
+                        };
+
+                        let mut exists = false;
+                        for (pos, col) in self.param.columns.iter().enumerate() {
+                            if col.name == column {
+                                partition_fields.push(
+                                    UnboundPartitionField::builder()
+                                        .source_id(pos as i32)
```

Comment:
I think we should enumerate iceberg_schema here and use the `field.is` instead of pos here.

Reply:
```
let id = match iceberg_schema.field_id_by_name(column) {
    Some(id) => id,
    None => bail!(format!("Invalid partition column: {}", column)),
};
```

Reply:
LGTM